IQSS/dataverse
# IQSS/9185 contact email updates

NOTE: per review, the settings have been replaced with Microprofile entries. The documentation has been updated but the PR notes below have not been updated!

**What this PR does / why we need it**: This PR includes several enhancements to the emails from the contact forms that have been implemented and deployed at QDR.  It also include fixes to make the /admin/feedback endpoint work.

Specific changes:
- new :CCSupportOnContactEmails setting which will include the support email address as a CC: entry when contact/feedback emails are sent to the contacts for a collection, dataset, or datafile. The setting also adds a CC line in the form so that users are aware that it will happen
- new :SupportEmail setting that allows a separate email, distinct from the :SystemEmail to be used as the from address in emails from the contact form/ feedback api. (Useful when admins want :SystemEmail to be a no-reply address. and/or to direct contact form emails to a different set of people/different support system.)
- Switch to sending one email to all contacts instead of sending an email to each. This is hopefully helpful in general (contacts can see who else is being informed) and simplifies adding a cc to support.
- Use of the PersonOrOrg algorithm to decide if a name refers to a person, in which case a \<lastname\>,\<firstname\> order should be reversed in the greeting ("Hello Homer Simpson"), or an Org (where a name with commas should not be changed).
- Addition of an algorithm to separate names with commas and include a final "and" when there are multiple contacts (e.g. "Hello Homer Simpson, Marge Simpson, and Bart Simpson")
- Update of tests to match
- Bug fixes/completion of the /admin/feedback endpoint so it sends emails.
- Docs for the api call and new settings, release notes.

**Which issue(s) this PR closes**:

Closes #9185

**Special notes for your reviewer**:

Depends on #9089 for the PersonOrOrgUtil changes. I've copied that class here so it will build.
Any of this can be broken out/dropped if the whole set of changes is not desired.

**Suggestions on how to test this**:
- Try the contact form and API call. Use multiple contacts for some tests to verify that one email is sent. 
- Try the CC setting and confirm there's a cc in the UI and the email. 
- Try the :SupportEmail setting and confirm that emails from the contact forms (and not other emails) are from that address.
- Try Org names like "IBM, Inc." to make sure they don't get flipped to "Hello Inc. IBM" in the greeting line in the emails.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

 Using the CCSupportOnContactEmails setting adds a line to the contact form: 
![image](https://user-images.githubusercontent.com/6731983/203611760-ada1b3ab-b857-491f-8f93-6f86bba07ad9.png)

**Is there a release notes update needed for this change?**:
included.

**Additional documentation**:
